pkgs/by-name/ba/basedpyright/package.nix

```
@@ -53,16 +55,49 @@ let
       runHook postInstall
     '';
   };
+
+  docify = python3.pkgs.buildPythonApplication {
+    pname = "docify";
+    version = "unstable";
+    format = "pyproject";
+    src = fetchFromGitHub {
+      owner = "AThePeanut4";
+      repo = "docify";
+      rev = "7380a6faa6d1e8a3dc790a00254e6d77f84cbd91";
+      hash = "sha256-BPR1rc/JzdBweiWmdHxgardDDrJZVWkUIF3ZEmEYf/A=";
+    };
+    buildInputs = [ python3.pkgs.setuptools ];
+    propagatedBuildInputs = [
+      python3.pkgs.libcst
+      python3.pkgs.tqdm
+    ];
+  };
+
+  docstubs = stdenvNoCC.mkDerivation {
+    name = "docstubs";
+    inherit src;
+    buildInputs = [ docify ];
+
+    installPhase = ''
+      runHook preInstall
+      cp -r packages/pyright-internal/typeshed-fallback docstubs
+      ${docify}/bin/docify docstubs/stdlib --builtins-only --in-place
+      cp -rv docstubs "$out"
+      runHook postInstall
+    '';
+  };
 in
 buildNpmPackage rec {
   pname = "basedpyright";
   inherit version src;
 
   sourceRoot = "${src.name}/packages/pyright";
-  npmDepsHash = "sha256-ZFuCY2gveimFK5Hztj6k6PkeTpbR7XiyQyS5wPaNNts=";
+  npmDepsHash = "sha256-wjwF1OlR9ohrl8gWW7ctVpeCq2Fu2m1XdHOEkXt7zjA=";
 
   postPatch = ''
     chmod +w ../../
+    mkdir ../../docstubs
+    cp -r ${docstubs}/stubs ../../docstubs
```

```suggestion
    cp -r ${docstubs}/stubs ../../docstubs
    ln -s ${docstubs}/stubs ../../docstubs
```
Will this work? Feel free to try it and it will avoid duplicating stuff in the store, plus it will look more coherent with the next commands. Also consider adding yourself to maintainers if you wish.

---

> Will this work? Feel free to try it and it will avoid duplicating stuff in the store, plus it will look more coherent with the next commands. Also consider adding yourself to maintainers if you wish.

It seems to work, though I don't know if this is a good practice...
What do you think @drupol ?

---

Because I fear that the source of the link could get GCed...

---

Seeing that it's a pattern that i see pretty often in nixpkgs judging by `rg -- 'ln -s \$\{.*\$out' pkgs`, would that be the case? I don't have the sufficient knowledge to confirm whether that should be the case but, as long as that source file is being referenced as a link, why should it be GCed?

---

I think it is fine indeed.

---

This should work fine, and GC is not a concern in the sense that, as @kiike says, nix guarantees referential integrity.

The only potential downsides I see:

- We now _can't_ garbage collect `${docstubs}` (since there's a reference), so if `${docstubs}` was significantly larger than `${docstubs}/stubs` we could actually win by copying. Judging by the name, this shouldn't matter much with `${docstubs}`, but this has been an issue e.g. with cuda static libraries
- If Nix is deployed on a network drive, symlinks can get expensive.

---

Thanks for this clarification @SomeoneSerge !